evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -102,6 +103,45 @@ def test_simple_imputer_col_with_non_numeric():
     assert_frame_equal(X_expected_arr, X_t, check_dtype=False)
 
 
+@pytest.mark.parametrize("data_type", ['pd', 'ww'])
+def test_imputer_all_bool(data_type):
+    X = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = pd.Series([1, 0, 0, 1, 0])
+    if data_type == 'ww':
+        X = ww.DataTable(X)
+        y = ww.DataColumn(y)
+    imputer = SimpleImputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+    X = pd.DataFrame([True, np.nan, False, np.nan, True], dtype=object)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype='category')
+    if data_type == 'ww':
+        X = ww.DataTable(X)
+    imputer = SimpleImputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
```

Comment:
This test is great!

I think the first two cases (i.e. above this line) are fully covered by the last case, right? I suggest you delete the first two cases, above this line, and keep the last one below.

Reply:
I'd argue for keeping all of the test cases. The first test case triggers the type conversion while the second one checks for proper boolean impution. The third one is more of an extension of the second one but it also combines a complete boolean column to make sure that the types are being processed properly.